pkg/config/structure/compatibility_test.go

```diff
@@ -91,7 +91,7 @@ network_devices:
 	})
 
 	warnings := viperConf.Warnings()
-	assert.Nil(t, warnings)
+	assert.Equal(t, &model.Warnings{TraceMallocEnabledWithPy2: false, Errors: []error(nil)}, warnings)
```

Comment:
Could you add a function `NewWarnings` to `pkg/config/model/warnings.go`, it should take an `[]error` to build the Warnings object. That way we can avoid exposing details about it here.